modal/_utils/match.py

```diff
@@ -0,0 +1,161 @@
+# Copyright Modal Labs 2024
```

Comment:
Maybe call this `docker_copy_match.py` or something less generic than `match` :)